src/nodes.coffee

```diff
@@ -3387,11 +3437,11 @@ exports.Assign = class Assign extends Base
   unfoldSoak: (o) ->
     unfoldSoak o, this, 'variable'
 
-  addScopeVariables: (o, {checkAssignability = yes} = {}) ->
+  addScopeVariables: (o, {allowAssignmentToExpansion = no} = {}) ->
```

Comment:
Had to refine the disabling of `checkAssignability` during AST generation to only specifically disable checking `Expansion`'s since it _should_ check assignability for things like `null` (eg `[null] = b`)